### DISCUSSION TITLE
Standards: Probability Distributions Software

### DISCUSSION THREAD

**Discussion by mpadge** [ropensci/statistical-software-review-book#58] [Standards]
This is a place to discuss our Standards for Probability Distributions Software which are under development. Community input and feedback is a critical part of the development process, and this discussion thread is the place for any suggestions for standards themselves, improvements, modifications, or edits small and large.

Participation in these discussions are guided by our [Code of Conduct](https://ropensci.org/code-of-conduct/).

We aim to develop standards for this category of *Probability Distributions Software* as openly as possible, and to document as much of the process as we can both here, and within the standards themselves, currently in [this pull request](https://github.com/ropensci/statistical-software-review-book/pull/59). As for all prior standards, the procedure will proceed something like this:

1. Collate list of *Probability Distributions* software
2. Use the software and note positive and negative aspects
3. Read the code and note positive and negative aspects
4. Convert results of the preceding two steps into general positive and negative aspects of *Probability Distributions* software.
5. Translate those general principles into standards

The entire process is expected to be iterative, with standards added, removed, and modified as the range of examined software is expanded, and hopefully in this case, as we gain more and more community feedback. Individual discussion items will reflect each of these stages, beginning with the list of useful software.

**Comment by mpadge** [comment]
## Probability Distributions Software

The reference collection on probability distributions software in R is given in the [CRAN Task View on *Probability Distributions*](https://cran.r-project.org/web/views/Distributions.html). Central to that *Task View* is the section on "*Unified Interface to Handle Distributions*" [sic], which includes or refers to the following packages:

- The [`distrXXX` family of packages](https://distr.r-forge.r-project.org/) (S4 classes)
- The [`distributional` package](https://github.com/mitchelloharawild/distributional) (S3 classes) by @mitchelloharawild
- The [`distributions3` package](https://github.com/alexpghayes/distributions3) (S3 classes) by @alexpghayes
- The [`distr6` package](https://github.com/alan-turing-institute/distr6/) (R6 classes) by @RaphaelS1

As in other categories, many of our standards will likely extend from recommendations either for the use of one or more of these "*United Interfaces*", or for alternative interfaces or interpretations which nevertheless follow general principles developed in these packages.

**Reply by grwhumphries** [reply]
@mpadge any particular reason why the base stats package isn't represented here? Allowing that would prevent from having to add more weight in packages that may already import a number of other packages.

**Reply by mpadge** [reply]
Mere oversight and presumption of implicit presence on my part. Could you please PR to emphasise at the outset that `stats` is the core package here?

**Comment by mpadge** [comment]
@alexpghayes @mitchelloharawild @RaphaelS1 @joethorley Your work has been formative in our [current draft of *Standards for Probability Distribution Software*](https://624c2c8da766c433b0f9747d--amazing-feynman-f84959.netlify.app/standards.html#standards-distributions) (temporary link 05 April 2022). (And @joethorley, your [`ssdtools` package](https://github.com/bcgov/ssdtools) ultimately doesn't get a mention only becuase it's domain-specific, but the code was very useful!) We would really appreciate any comments, insights, suggestions from any of you, and indeed from anybody else at all, particularly in response to these questions:

1. Do you agree with the standards as currently written?
2. Do you have any comments on  likely generality or usefullness of the standards?
3. Can you think of any potential standards we may have missed?

**Comment by joethorley** [comment]
Thanks @mpadge - I'll review when I have sometime. Note did you review the code in our latest release v1.0.0 which was merged from the dev branch into the main branch on the 1st of April?

**Reply by mpadge** [reply]
Ah, nope, I didn't review that code, but had a look through the PR now - thanks!

**Reply by joethorley** [reply]
@mpadge - I finally found sometime to review. Much of the standards are outside my area of expertise - I'm a biologist turned programmer and modeler as opposed to a pure statistician. That said I thought it was wise to leave the fitting part open as its such a big area and that your recommendations for testing were good.

**Reply by mpadge** [reply]
Thanks @joethorley for the useful and affirmative feedback. (And no disclaimers necessary here: I'm also, coincidentally, an ecologist turned programmer and modeller.)
